openlibrary/plugins/openlibrary/js/merge-request-table/index.js

```diff
@@ -169,115 +160,120 @@ export function initShowAllCommentsLinks(elems) {
 
 /**
  * Toggles visibility of a request's older comments.
- *
- * @param {HTMLELement} elem Element which contains a reference to the old comments
- */
+*
+* @param {HTMLELement} elem Element which contains a reference to the old comments
+*/
 function toggleAllComments(elem) {
-    const targetId = elem.dataset.targetId;
-    const target = document.querySelector(`#${targetId}`)
-    target.classList.toggle('hidden')
-
-    const isHidden = target.classList.contains('hidden')
-    const prevSibling = elem.previousElementSibling;
-    if (isHidden) {
-        prevSibling.textContent = 'Showing most recent comment only.'
-        elem.textContent = 'View all'
-    } else {
-        prevSibling.textContent = 'Showing all comments.'
-        elem.textContent = 'View most recent only'
-    }
+    const targetHiddenComments = elem.dataset.hiddenComments;
+    const targetLatestComment = elem.dataset.latestComment || 0;
+    const targetIdOldComments = elem.dataset.oldComments;
+    const targetBtnComments = elem.dataset.btnComments;
+
+    const hiddenCommentsTarget = document.querySelector(`#${targetHiddenComments}`)
+    const latestCommentTarget = document.querySelector(`#${targetLatestComment}`)
+    const oldCommentsTarget = document.querySelector(`#${targetIdOldComments}`)
+    const targetCommentsBtn = document.querySelector(`.${targetBtnComments}`);
```

Comment:
This would be better if we weren't querying the document for these elements every time this function is called.